docs/releases/1.10.3.txt

```diff
@@ -23,3 +23,6 @@ Bugfixes
 
 * Fixed ``QuerySet.bulk_create()`` on PostgreSQL when the number of objects is
   a multiple plus one of ``batch_size`` (:ticket:`27385`).
+
+* Fixed redirecting issue when session destroyed during request processing
```

Comment:
That was along the lines of my thinking. Unfortunately, there's no way to trigger `handler400` besides `SuspiciousOperation`. An alternative could be to return an `HttpResponse(status=400)` (with what content?) in the middleware rather than raising the exception. Another option could be to go with the current approach for now and create a ticket to try to improve it in master, perhaps with some new exception class.